modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,404 @@
+  /// lifetime in [UserProfileConfig], or disable it on a call to call basis by
+  /// setting [useCache] to false.
+  ///
+  /// Throws a [UserProfileNotFoundException] in case no profile exists for the given [authUserId].
+  static Future<UserProfileModel> findUserProfileByUserId(
+    final Session session,
+    final UuidValue authUserId, {
+    final bool useCache = true,
+  }) async {
+    final profile = await maybeFindUserProfileByUserId(
+      session,
+      authUserId,
+      useCache: useCache,
+    );
+
+    if (profile == null) {
+      throw UserProfileNotFoundException(authUserId);
+    }
+
+    return profile;
+  }
+
+  /// Looks for a user profile by the `AuthUser`'s ID.
+  ///
+  /// Returns `null` if no profile is found. By default the
+  /// result is cached locally on the server. You can configure the cache
+  /// lifetime in [UserProfileConfig], or disable it on a call to call basis by
+  /// setting [useCache] to false.
+  static Future<UserProfileModel?> maybeFindUserProfileByUserId(
+    final Session session,
+    final UuidValue authUserId, {
+    final bool useCache = true,
+  }) async {
+    final cacheKey = _userProfileCacheKey(authUserId);
+    UserProfileModel? userInfo;
+
+    if (useCache) {
+      userInfo = await session.caches.local.get<UserProfileModel>(cacheKey);
+      if (userInfo != null) return userInfo;
+    }
+
+    final userProfile = await _maybeFindUserProfile(session, authUserId);
+
+    if (userProfile != null) {
+      userInfo = userProfile.toModel();
+    }
+
+    if (useCache && userInfo != null) {
+      await session.caches.local.put(
+        cacheKey,
+        userInfo,
+        lifetime: UserProfileConfig.current.userInfoCacheLifetime,
+      );
+    }
+
+    return userInfo;
+  }
+
+  /// Updates a profiles's user name.
+  static Future<UserProfileModel> changeUserName(
+    final Session session,
+    final UuidValue authUserId,
+    final String? newUserName,
+  ) async {
+    final userProfile = await _findUserProfile(
+      session,
+      authUserId,
+    );
+
+    userProfile.userName = newUserName;
+
+    final updatedProfile = await _updateProfile(session, userProfile);
+
+    return updatedProfile.toModel();
+  }
+
+  /// Updates a profile's full name.
+  static Future<UserProfileModel> changeFullName(
+    final Session session,
+    final UuidValue authUserId,
+    final String? newFullName,
+  ) async {
+    final userProfile = await _findUserProfile(
+      session,
+      authUserId,
+    );
+
+    userProfile.fullName = newFullName;
+
+    final updatedProfile = await _updateProfile(session, userProfile);
+
+    return updatedProfile.toModel();
+  }
+
+  /// Remove the user profile for the given [authUserId].
+  ///
+  /// In case the user did not have a profile, nothing is changed.
+  static Future<void> deleteProfileForUser(
+    final Session session,
+    final UuidValue authUserId,
+  ) async {
+    await UserProfile.db.deleteWhere(
```

Comment:
TODO:
Delete images from storage

Reply:
Updated the implementation to also (try to) delete images from storage.

If the deletion from storage fails, only an error is logged, as we can not recover from that, and keeping the `UserProfileImage` with a pointer to a file that is broken is likely not a better course of action.